vnext/include/ReactUWP/IReactInstance.h

```diff
@@ -50,7 +50,11 @@ struct ReactInstanceSettings {
   std::string BundleRootPath;
   facebook::react::NativeLoggingHook LoggingCallback;
   std::function<void(facebook::react::JSExceptionInfo &&)> JsExceptionCallback;
+#if defined(USE_HERMES)
+  JSIEngine jsiEngine{JSIEngine::Hermes};
```

Comment:
I'm not sure that I agree with this change. We have a similar flag for V8. What would we do if both were enabled at the same time?

If anything, I'd almost rather just rename "USE" to "WITH" to make intent clearer, that it enables usage instead of changing default. 

@tudor-ms will have more context on the right thing to do here.

Reply:
Right now USE_V8 is always turned on, if its not arch arm. I do agree there is some issue with this.
I had opened an issue to formally address it since I really am not a fan of preprocessor macros even controlling any of this. 
#4288

Reply:
Some of the build logic related stuff is needed because we will actually dynamically switch out headers. I don't believe the intent is to use them in code for selection.

Reply:
So if instead to pivot the conversation slightly ,  USE_V8 isn't doing anything that effects the consumer of RNW and you have no way to turn on using v8, should this be changed to some macro where I can enable jsiEngine for v8 then?

Reply:
Do you know if V8 isnt supported on arm?

Reply:
Backing up a bit, could I get a better understanding of your intended use of Hermes and V8? There's still quite a bit or work to get them to ship quality, especially on UWP. The current support has been present mostly for prototyping and development. I'm not sure we've done a great job in communicating this.

@tudorms will have context on readiness for V8, but as a forwarning a lot of the initial development for V8 and Hermes has been targeting Win32 Office as a first stop.

Reply:
Copy/paste from similiar/related thread in #4288 


"We need to support arm and arm64. And when we asked about some ChakraCore. I got the impression it wasn't really the suggested path forward.
> "I’m a little surprised there’s a need for ChakraCore. It’s not the path forward now that we’ve moved to Chromium."

I got the impression it was deprecated and not having any more resources allocated to it.

From what I can gather V8 isn't for Windows arm/arm64. So that leaves us with Hermes which seems to support all platforms. And we are prepping to start shipping to rest of MS in the coming weeks and start expanding our audience rapidly.  If v8 supports arm/arm64 on windows then that'd be fine with us.

Also we are interested in the small perf wins of either v8 or hermes over regular Chakra."